pkg/operator/controller/externaldns/status.go

```diff
@@ -0,0 +1,296 @@
+			// Sort keys so that the result is deterministic.
+			keys := make([]*corev1.Pod, 0, len(unscheduled))
+			for pod := range unscheduled {
+				keys = append(keys, pod)
+			}
+			sort.Slice(keys, func(i, j int) bool {
+				if keys[i].CreationTimestamp.Equal(&keys[j].CreationTimestamp) {
+					return keys[i].UID < keys[j].UID
+				}
+				return keys[i].CreationTimestamp.Before(&keys[j].CreationTimestamp)
+			})
+			for _, pod := range keys {
+				cond := unscheduled[pod]
+				if cond.Reason == corev1.PodReasonUnschedulable {
+					haveUnschedulable = true
+					message += fmt.Sprintf("%s Pod %q cannot be scheduled: %s", message, pod.Name, cond.Message)
+				} else {
+					message += fmt.Sprintf("%s Pod %q is not yet scheduled: %s: %s", message, pod.Name, cond.Reason, cond.Message)
+				}
+			}
+			if haveUnschedulable {
+				message = message + " Make sure you have sufficient worker nodes."
+			}
+			return metav1.Condition{
+				Type:    ExternalDNSPodsScheduledConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  "PodsNotScheduled",
+				Message: message,
+			}
+		}
+		return metav1.Condition{
+			Type:    ExternalDNSPodsScheduledConditionType,
+			Status:  metav1.ConditionTrue,
+			Reason:  "AllPodsScheduled",
+			Message: "All pods are scheduled",
+		}
+	}
+
+}
+
+// MergeConditions updates the conditions list with a new condition
+// the condition is added if no conditions of the same type exists
+// otherwise, the condition is merged with the existing condition of the same type
+func mergeConditions(conditions []metav1.Condition, updates ...metav1.Condition) []metav1.Condition {
+	now := metav1.NewTime(clock.Now())
+	for i, update := range updates {
+		add := true
+		for j, cond := range conditions {
+			if cond.Type == update.Type {
+				add = false
+				if conditionChanged(cond, update) {
+					conditions[j] = update
+					conditions[j].LastTransitionTime = now
+					break
+				}
+			}
+		}
+		if add {
+			updates[i].LastTransitionTime = now
+			conditions = append(conditions, updates[i])
+		}
+	}
+	return conditions
+}
+
+func conditionChanged(a, b metav1.Condition) bool {
+	return a.Status != b.Status || a.Reason != b.Reason || a.Message != b.Message
+}
+
+func createDeploymentAvailabilityUnknownCondition() metav1.Condition {
+	return metav1.Condition{
+		Type:               ExternalDNSDeploymentAvailableConditionType,
+		Status:             metav1.ConditionUnknown,
+		Reason:             "DeploymentAvailabilityUnknown",
+		Message:            "The deployment has no Available status condition set",
+		LastTransitionTime: metav1.NewTime(clock.Now()),
+	}
+}
+
+// updateExternalDNSStatus recomputes all conditions given the current deployment and its status
+// and pushes the new externalDNS custom resource with updated status through a call to the client.Update function
+func (r *reconciler) updateExternalDNSStatus(ctx context.Context, externalDNS *operatorv1alpha1.ExternalDNS, currentDeployment *appsv1.Deployment) error {
```

Comment:
Can you please move this method to the top of the file?
Would be a little more readable as it's the main function of the file which calls all the others.